File: week-01/yc_1874_stack_sequence.cs
using System.Text;

namespace ProblemSolving.week01;

public class Yc1874
{
    static void Main(string args[])
    {
        int stackSize = int.Parse(Console.ReadLine());
        StringBuilder answer = new StringBuilder();
        bool resultAvailable = true;
        int valueToPush = 1;
        Stack<int> stack = new Stack<int>();

        for (int i = 0; i < stackSize; i++)
        {
            int currentNumber = int.Parse(Console.ReadLine());

            while (valueToPush <= currentNumber)
            {
                stack.Push(valueToPush++);
                answer.Append("+").Append("\n");
            }

            if (stack.Count > 0 && stack.Peek() == currentNumber)
            {
                stack.Pop();
                answer.Append("-").Append("\n");
            }
            else
            {
                resultAvailable = false;
            }
        }

        if (resultAvailable)
        {
            Console.WriteLine(answer);
        }
        else
        {
            Console.WriteLine("NO");
        }
    }
}

File: week-01/yc_1874_stack_sequence.java
import java.util.Stack;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Yc1874 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int stackSize = Integer.parseInt(br.readLine());
        StringBuilder answer = new StringBuilder();
        boolean resultAvailable = true;
        int valueToPush = 1;
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < stackSize; i++) {
            int currentNumber = Integer.parseInt(br.readLine());

            while (valueToPush <= currentNumber) {
                stack.push(valueToPush++);
                answer.append("+\n");
            }

            if (stack.peek() == currentNumber) {
                stack.pop();
                answer.append("-\n");
            } else {
                resultAvailable = false;
            }
        }

        if (resultAvailable) {
            System.out.println(answer);
        } else {
            System.out.println("NO");
        }
    }
}